blog/2023-12-12-magazine-007.md

```diff
@@ -0,0 +1,291 @@
+CLI.
+
+First, install `ts4w` (temporary name) which includes some utility functions for Wing:
+
+```sh
+npm i ts4w
+```
+
+Now, create a `main.ts` file:
+
+```js
+import { main } from "ts4w";
+import { cloud } from "@winglang/sdk";
+
+main(app => {
+  new cloud.Bucket(app, "MyBucket");
+});
+```
+
+Wing It:tm::
+
+```sh
+wing it main.ts
+```
+
+This will open up [Wing Console](https://www.winglang.io/docs/tools/wing-console) with your program.
+
+The next step is to enable consuming [winglibs](https://github.com/winglang/winglibs) such as
+[websockets](https://github.com/winglang/winglibs/tree/main/websockets) from TypeScript. Stay tuned
+and let us know what you think!
+
+❤️ [Mark McCulloh](https://github.com/MarkMcCulloh)
+
+### Built-in support for `.env` files
+
+[`.env` files](https://www.npmjs.com/package/dotenv) are now automatically loaded across all CLI
+workflows (`run`, `test`, `compile`). 
+
+> Environment variables can be consumed only *during preflight* via `util.env("VAR")`, or `tryEnv()`
+> for the risk-averse of us.
+
+We also support interpolation:
+
+```sh
+BASE_URL=https://www.winglang.io
+API_BASE_URL=${BASE_URL}/api
+```
+
+You can also override environment variables based on the the CLI subcommand:
+
+- `.env` - Always loaded
+- `.env.[mode]` - Only loaded in the specified mode (eg. `run`, `compile`, `test`, etc.)
+
+❤️ [Gary Sassano](https://github.com/garysassano)
+
+### GCP support in `wing test`
+
+The Wing Test Framework is one of the coolest things about Wing. It allows developers to write tests
+that cover **both infrastructure and runtime code** and can run across multiple cloud platforms.
+
+Now that Google Cloud Platform (GCP) is supported, `wing test` supports all built-in Terraform
+platforms (AWS, Azure and GCP).
+
+For example, the following Wing program includes a test for the `store()` function:
+
+```js
+bring cloud;
+bring util;
+bring expect;
+
+let bucket = new cloud.Bucket();
+
+let store = inflight (data: str) => {
+  bucket.put("data.txt", data);
+};
+
+test "store puts data into the bucket" {
+  let data = "my data " + util.uuidv4();
+  store(data);
+
+  expect.equal(bucket.get("data.txt"), data);
+}
+```
+
+Now, we can run this test within the Wing Simulator (the `sim` target):
+
+```console
+$ wing test main.w
+pass ─ main.wsim » root/env0/test:store puts data into the bucket
+
+
+Tests 1 passed (1)
+Test Files 1 passed (1)
+Duration 0m1.23s
+```
+
+The `-t` flag can be used to compile and deploy this test to the target platform:
+
+```console
+$ wing test -t tf-gcp main.w
+✔ Compiling main.w to tf-gcp...
+⠋ terraform init
+...
+```
+
+❤️ [Tsuf Cohen](https://github.com/tsuf239)
+
+### UX improvements to Wing Console
+
+We've made some improvements to how Wing Console visualizes connections between nodes.
+
+Before:
+
+![console-connections-cleanup-before](./assets/console-connections-cleanup-before.png)
```

Comment:
We need to verify these render correctly when we send the email. Last time we had some issues.

Reply:
Will do.